cds-snc/forms-terraform
# Terraform provider and AWS upgrade to

# Summary | Résumé
AWS and Terraform provider upgrade draft


# Test instructions:

1. (optional) install `tfenv` to help manage terraform version with `brew install tfenv` and `tfenv use 1.5.0`  [usage doc](https://github.com/tfutils/tfenv#usage) for more details.
2. In your console import import AWS credentials from this `RaphaelKeita-scratch` or you may request a scratch account (contact core team or patheard).
3. `export AWS_ACCOUNTID and APP_ENV=[Name]_scratch`
4. `export AWS_SECRET_ACCESS_KEY=xxx   export AWS_SESSION_TOKEN=XXXX`
5. Create the  satellite bucket with AWS_ACCOUNTID.
6. Create `.env` and add all required environment variables values 
7. cd into `aws/app` and run ` terragrunt apply` 
8. (optional) _From `env/cloud`  in each of the following folders `Kms, Network, rds, redis, dynamoDB, sqs, hostedzone, ecr, app`  `terragrunt init -upgrade` and `terragrunt apply` .  Caution Before deploying the app module make sure to  pull and push an app docker image into the `ECR` repository (slack me)


# Remarks
During the deployment of the lambda infrastructure, a problem was identified. Upon investigation, it was found that the V4.9.0 version of the terraform aws provider does not yet support Node.js v16, which is the current version in staging. The solution was to change the Node.js runtime version to v14 for lambdas functions. 

Acceptance Criteria:
- [x] Terraform is upgraded to V1.5.0
- [x] AWS terraform provider upgraded to at least V4.9.0
- [x] AWS and Terraform provider versions have been upgraded to the latest stable releases.
- [x]  Existing infrastructure and codebase/configurations are compatible with the upgraded providers.
- [x] Comprehensive testing has been conducted to verify the proper functioning of the upgraded providers.
